homeassistant/components/ozw/light.py

```diff
@@ -42,23 +61,80 @@ def byte_to_zwave_brightness(value):
     return 0
 
 
-class ZwaveDimmer(ZWaveDeviceEntity, LightEntity):
-    """Representation of a Z-Wave dimmer."""
+# Create ZwaveLight class and combine them
+class ZwaveLight(ZWaveDeviceEntity, LightEntity):
+    """Representation of a Z-Wave light."""
 
     def __init__(self, values):
         """Initialize the light."""
         super().__init__(values)
+        self._color_channels = None
+        self._hs = None
+        self._ct = None
+        self._white = None
         self._supported_features = SUPPORT_BRIGHTNESS
         # make sure that supported features is correctly set
         self.on_value_update()
 
     @callback
     def on_value_update(self):
         """Call when the underlying value(s) is added or updated."""
-        self._supported_features = SUPPORT_BRIGHTNESS
         if self.values.dimming_duration is not None:
             self._supported_features |= SUPPORT_TRANSITION
 
+        if self.values.color is not None:
+            self._supported_features |= SUPPORT_COLOR
+
+        if self.values.color_channels is not None:
+            self._supported_features |= SUPPORT_WHITE_VALUE
+
+        if self.values.color is None:
+            return
+        if self.values.color_channels is None:
```

Comment:
Is color channels needed also for RGB lights not only for RGBW lights?

Reply:
From what I understand, if you have just RGB you will also still have color_channels. They go hand in hand.

Reply:
So then we can simplify these checks by combining them?

Reply:
Combined in latest commit.